core/src/nucleus/ribosome/memory.rs

```diff
@@ -16,16 +16,19 @@ pub struct SinglePageManager {
     wasm_memory: MemoryRef,
 }
 
-/// A Memory Manager limited to one memory page that works like a stack
-/// With this Memory Manager, Host and WASM pass around only a i32.
-/// That i32 is the last memory allocation on the stack: a i16 offset and a i16 length
-/// (which fits with the 64KiB sized of a memory Page).
-/// Complex Input arguments should be stored on the latest allocation on the stack.
-/// Complex Output arguments can be stored anywhere on stack.
-/// ErrorCode passing is also made possible by convention:
-/// using i16 offset as error code and i16 length to zero to indicate its an error code.
-///
-/// In the future we could do same with i32 -> i64 and handle multiple memory Pages.
+/// A Memory Manager limited to one wasm memory page that works like a stack.
+/// With this Memory Manager, Host and WASM only need to pass around an i32 to communicate any data.
+/// That i32 is the last memory allocation on the stack:
+/// it is split in an i16 'offset' in th upper bits and an i16 'length' in the lower bits.
```

Comment:
typo s/th/the/

Reply:
Thanks. Next time please edit directly in github.